projects/soft_999_Jenish_Patel_fw_102/src/main.c

```diff
@@ -0,0 +1,44 @@
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define HALF_SECOND_MS 500  //
+
+typedef struct CounterStorage {
+  uint16_t counter_a;
+  uint16_t counter_b;
+} CounterStorage;
+
+void prv_timer_call(const SoftTimerId timer_id, void *context) {
+  // cast void* to our struct so we can use it
+  CounterStorage *storage = context;
+  storage->counter_a++;
+  storage->counter_b++;
+
+  // Prints Every half-second
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a);
+
+  // Prints Every Second
+  if (storage->counter_b % 2 == 0) {
+    LOG_DEBUG("Counter B: %i\n", storage->counter_b / 2);
+  }
+
+  // start the timer again, so it keeps periodically flipping coins
+  soft_timer_start_millis(HALF_SECOND_MS, prv_timer_call, &storage, NULL);

```

Comment:
I think you have an error here with the type of `storage`, check the parameters that `soft_timer_start` takes. Also check your output (run it locally) to check for errors.